From 9be7c5a98b0685a60158aff19a5d472e64b2057a Mon Sep 17 00:00:00 2001
From: Artem Borisov <dedsa2002@gmail.com>
Date: Thu, 16 Aug 2018 22:17:21 +0300
Subject: [PATCH 08/12] HACK: SF: Force client composition for all layers

For unknown reasons using device composition on all
kirin devices introduces significant image quality drop.
Let's force client composition (no HW overlays) until
we figure the proper solution.

Change-Id: Iccec69e66f27d2d6eff55fc9cb6e5c0e1ef515c5
---
 services/surfaceflinger/SurfaceFlinger.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 87baf8c8a..cca2e98a8 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -2008,9 +2008,9 @@ void SurfaceFlinger::setUpHWComposer() {
                     }
 
                     layer->setGeometry(displayDevice, i);
-                    if (mDebugDisableHWC || mDebugRegion) {
+                    //if (mDebugDisableHWC || mDebugRegion) {
                         layer->forceClientComposition(hwcId);
-                    }
+                    //}
                 }
             }
         }
-- 
2.17.1

